R/utils.R

```diff
@@ -0,0 +1,32 @@
+fromRJSON <- function(x, ...) {
+  return(jsonlite::fromJSON(sprintf(x, ...),
+    simplifyVector = FALSE,
+    simplifyDataFrame = FALSE,
+    simplifyMatrix = FALSE
+  ))
+}
+
+handle_err_result <- function(res) {
+  if (res$type == "error") {
+    return(TRUE)
+  }
+  return(FALSE)
+}
+
+next_id <- 0
+get_new_id <- function() {
+  # next_id <<- next_id + 1
+  return(next_id)
+}
+
+get_filetoken <- function(filepath = NULL, content = NULL) {
+  if (!is.null(filepath)) {
+    if (!file.exists(filepath)) {
+      stop("File does not exist: ", filepath)
+    }
+    token <- digest::digest(readLines(filepath))
+  } else if (!is.null(content)) {
+    token <- digest::digest(content)
+  }
+  return(token)
```

Comment:
I don't think we even use file tokens for the VSCode extension so this is a very nice feature!!

Reply:
Instead of checking `!is::null` it would probably be better to verify the type with something like `is::character`. 
In general we may want to add [documentation](https://roxygen2.r-lib.org/) to such functions.

Reply:
While I agree that documenting this function would be helpful, I don't think that the `is.null` check should be replaced by an `is.character` check.

Passing `NULL` explicitly or implicitly to, e.g., the `filepath` parameter means that it's absent, while passing a number is just a wrong input. In this case, we should not try to go ahead with the value passed to `content`, but fail directly.

IMHO something like this would be better than just replacing `is.null` by `is.character`
```r
get_filetoken <- function(filepath = NULL, content = NULL) {
  if (!is.null(filepath)) {
    if (!is.character(filepath) || !file.exists(filepath)) {
      stop("File does not exist: ", filepath)
    }
    token <- digest::digest(readLines(filepath))
  } else if (!is.null(content)) {
    if (!is.character(content)) {
       stop("Pls. provide a string")
    }
    token <- digest::digest(content)
  }
  return(token)
}
```

Reply:
In that sense we could check with `missing`. Then again, blocking with `stopifnot` is fine (and was my original intention).

Reply:
Resolved in 569eb17